src/interface/blas1_interface.hpp

```diff
@@ -246,6 +246,108 @@ typename sb_handle_t::event_t _asum_impl(
 }
 #endif
 
+/**
+ * _iamax_iamin_impl.
+ * Internal implementation of backend specific iamax or iamin operator.
+ * This method launches up to 2 instances of IndexMaxMin class
+ * to compute the integer index of max or min value in the input. Based
+ * on the localMemSize template parameter, the implementation performs
+ * work group reduction using local memory or sub_group reduction
+ * using shuffle operation to compute the output.
+ *
+ * @tparam localSize value to indicate work group size
+ * @tparam localMemSize value to indicate size of local memory required
+ * @tparam is_max boolean variable to indicate if required operation is
+ * iamax or not
+ * @tparam single boolean variable to indicate whether to execute a single
+ * step reduction or a two step reduction
+ * @tparam sb_handle_t SB_Handle type
+ * @tparam container_0_t Buffer Iterator or USM pointer
+ * @tparam container_1_t Buffer Iterator or USM pointer
+ * @tparam index_t Index type
+ * @tparam increment_t Increment type
+ *
+ * @param sb_handle sb_handle
+ * @param _N size of the input vector
+ * @param _vx input vector
+ * @param _incx Stride of vector x (i.e. measured in elements of _vx)
+ * @param _rs output variable
+ * @param _nWG no. of work groups required for the reduction
+ * @param _dependencies Vector of events
+ */
+
+template <int localSize, int localMemSize, bool is_max, bool single,
+          typename sb_handle_t, typename container_0_t, typename container_1_t,
+          typename index_t, typename increment_t>
+typename sb_handle_t::event_t _iamax_iamin_impl(
+    sb_handle_t &sb_handle, index_t _N, container_0_t _vx, increment_t _incx,
+    container_1_t _rs, const index_t _nWG,
+    const typename sb_handle_t::event_t &_dependencies) {
+  typename VectorViewType<container_0_t, index_t, increment_t>::type vx =
+      make_vector_view(_vx, _incx, _N);
+  auto rs = make_vector_view<index_t>(_rs, static_cast<increment_t>(1),
+                                      static_cast<index_t>(1));
+  auto tupOp = make_tuple_op(vx);
+  typename sb_handle_t::event_t ret;
+  if constexpr (single) {
+    auto op = make_index_max_min<is_max, false>(rs, tupOp);
+    if constexpr (localMemSize == 0) {
+      auto q = sb_handle.get_queue();
+      const index_t sg_size = static_cast<index_t>(
+          q.get_device()
+              .template get_info<sycl::info::device::sub_group_sizes>()[0]);
+      ret = sb_handle.execute(op, sg_size, sg_size, _dependencies);
+    } else {
+      ret =
+          sb_handle.execute(op, static_cast<index_t>(localSize),
+                            _nWG * static_cast<index_t>(localSize),
```

Comment:
Thanks for your patch, @muhammad-tanvir-1211 .

I think that `_nWG` must be `1` since `single` is true or I'm missing something.

In that case, I would add a comment explaining it and 
```suggestion
                            static_cast<index_t>(localSize),
```

Reply:
Fixed this in the recent commit.